### PR TITLE
meta: add express 5 as peer dependency

### PR DESCRIPTION
**Pull request by InSantoshMahto**
## What Does This PR Do?
This PR will resolve the peer dependency error fix for express v5

## Checklist

- [ ] The issues that this PR fixes/closes have been mentioned above.
- [ ] What this PR adds/changes/removes has been explained.
- [ ] All tests (`npm test`) pass.
- [ ] All added/modified code has been commented, and
      methods/classes/constants/types have been annotated with TSDoc comments.
- [ ] If a new feature has been added or a bug has been fixed, tests have been
      added for the same.

### CODE REVIEW

**Comment by gamemaker1** [comment]
Hi @InSantoshMahto,

Thanks for taking the time to make the PR :)

However, Express v5 is still in beta - I think it would make more sense to update the peer dependency once v5.0.0 is officially released and tested with this library.

**Comment by gamemaker1** [comment]
Also, why does this change involve 57 files? Did you need to change something to make it work with Express v5?

**Comment by InSantoshMahto** [comment]
> Also, why does this change involve 57 files? Did you need to change something to make it work with Express v5?

I just rename the source directory to src to look the same as other projects. and remove some comments

File: package.json
Comment (gamemaker1) [inline package.json:3]: Please don't bump the version, we will do that as part of the release process later on

**Comment by InSantoshMahto** [comment]
> Hi @InSantoshMahto,
> 
> Thanks for taking the time to make the PR :)
> 
> However, Express v5 is still in beta - I think it would make more sense to update the peer dependency once v5.0.0 is officially released and tested with this library.

Can we make it available express-rate-limit@next so that we can use with express@next for V5

**Comment by gamemaker1** [comment]
> Can we make it available express-rate-limit@next so that we can use with express@next for V5

I think we could do that. AFAIK usage with Express v5 should be possible even if the peer dependency is not updated though - npm only warns if peer dependencies are not met. Have you tested the library and checked if it works with the newer version without any changes?

**Comment by gamemaker1** [comment]
> I just rename the source directory to src to look the same as other projects. and remove some comments

Could you please undo those changes? The comments are helpful for new contributors and I prefer keeping full form directory names in projects. Also, the tests are failing because of the incomplete rename.

**Comment by InSantoshMahto** [comment]
> > I just rename the source directory to src to look the same as other projects. and remove some comments
> 
> Could you please undo those changes? The comments are helpful for new contributors and I prefer keeping full form directory names in projects. Also, the tests are failing because of the incomplete rename.

sure

**Comment by InSantoshMahto** [comment]
> > Can we make it available express-rate-limit@next so that we can use with express@next for V5
> 
> I think we could do that. AFAIK usage with Express v5 should be possible even if the peer dependency is not updated though - npm only warns if peer dependencies are not met. Have you tested the library and checked if it works with the newer version without any changes?

Yes. work perfectly with waring while using npm. But while using pnpm I faced the issue. it break my docker build.

File: package.json
Comment (InSantoshMahto) [reply package.json]: Sure

**Comment by gamemaker1** [comment]
> But while using pnpm I faced the issue. it break my docker build.

What error do you receive?

**Comment by InSantoshMahto** [comment]
> > But while using pnpm I faced the issue. it break my docker build.
> 
> What error do you receive?
> 
> In a test project, running `pnpm install express@5.0.0-beta.1` gives me the following output:
> 
> ```
> Packages: +13 -12
> +++++++++++++------------
> 
> Progress: resolved 295, reused 268, downloaded 8, added 13, done
> 
> dependencies:
> - express 4.17.3
> + express 5.0.0-beta.1
> - express-rate-limit 6.3.0
> + express-rate-limit 6.3.0
> 
>  WARN  Issues with peer dependencies found
> .
> └─┬ express-rate-limit
>   └── ✕ unmet peer express@^4: found 5.0.0-beta.1
> ```

yes same error I also received

**Comment by gamemaker1** [comment]
This is just a warning, and shouldn't result in a failure during installation.

Anyways, I don't see an issue in publishing a new version under the `next` tag. @nfriedly what do you think?

**Comment by nfriedly** [comment]
I think express 5.x compatibility is fine to declare in a regular semver minor release.

I would like to see us run our automated tests against both 4.x and 5.x then - but I think that could be as simple as adding a few lines to [test/external/run-all-tests](https://github.com/nfriedly/express-rate-limit/blob/master/test/external/run-all-tests) - basically, after the two `npm test` lines, do `npm install express@5` and then `npm test` again.

**Comment by gamemaker1** [comment]
@InSantoshMahto could you please check the "allow maintainer edits" box on the right hand side on this page so I can push a commit updating the tests? Thanks!

**Comment by InSantoshMahto** [comment]
> @InSantoshMahto could you please check the "allow maintainer edits" box on the right hand side on this page so I can push a commit updating the tests? Thanks!
Screenshot 2022-04-24 at 1 55 21 PM
Hey @gamemaker1 Already checked

**Comment by gamemaker1** [comment]
Ohk, thank you

**Review by gamemaker1 — APPROVED** [review]
Looks good to me!

**Comment by gamemaker1** [comment]
@nfriedly Do take a look at the changes, the tests now run with express 4 and 5

**Review by nfriedly — APPROVED** [review]
I like it!

it might be good to add a couple of `echo "running test $foo"; npm ls express` sort of comments to `run-all-tests`, but that's more cosmetic than anything else.

**Comment by gamemaker1** [comment]
Sure, will do

File: package.json
Comment (nfriedly) [inline package.json:82]: oh, I think this should probably stay on 4.x if it's going to be the starting point for tests that first run against 4.x, then5.x

Comment (gamemaker1) [reply package.json:82]: This version is actually not changed during the tests - its only the version of express used by the projects that changes. This express dependency is only for type definitions, and AFAIK type defs for 5.0.0 haven't been released (or maybe they are the same as 4.x?)

Comment (gamemaker1) [reply package.json:82]: Perhaps this should stay at 4.17 then

Comment (nfriedly) [reply package.json:82]: yeah, I guess it doesn't matter too much, but let's keep it at 4.x for now.

**Comment by gamemaker1** [comment]
Added some colors 🎆

**Comment by gamemaker1** [comment]
Merging after fixing a merge conflict, will release a minor version by tomorrow :D

**Comment by gamemaker1** [comment]
Released as part of [v6.4.0](https://github.com/nfriedly/express-rate-limit/releases/tag/v6.4.0)

**Pull request merged**
